File: unidad14/ListadoPersonas/src/domain/interfaces/repositories/IRepositoryPersonas.ts
import { Persona } from '../../entities/Persona';

export interface IRepositoryPersonas {
  getListaPersonas(): Promise<Persona[]>;
  crearPersona(personaNueva: Persona): Promise<number>;
  actualizarPersona(idPersona: number, persona: Persona): Promise<number>;
  eliminarPersona(idPersona: number): Promise<number>;
  getPersonaById(idPersona: number): Promise<Persona | null>;
  contarPersonaDepartamento(idDepartamento: number): Promise<number>;
}

File: ListadoPersonasYDepartamentos/Domain/Repositories/IRepositoryPersonas.cs
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Repositories
{
    public interface IRepositoryPersonas
    {
        Persona[] getListaPersonas();

        int crearPersona(Persona personaNueva);

        int actualizarPersona(int idPersona, Persona persona);

        int eliminarPersona(int idPersona);

        Persona getPersonaById(int idPersona);

        int contarPersonadepartamento(int idDepartamento);


    }
}
